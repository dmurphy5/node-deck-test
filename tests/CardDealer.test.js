const CardDealer = require("../CardDealer");

const mockCustomDeck = [
    { color: "red", value: 1 },
    { color: "green", value: 2 },
    { color: "blue", value: 2 }
];

describe("CardDealer", () => {
    beforeEach(() => jest.clearAllMocks());
    test("Creates a standard deck if no deck is passed", () => {
        const spy = jest.spyOn(CardDealer.prototype, "_createStandardDeck");
        const dealer = new CardDealer();

        expect(spy).toHaveBeenCalled();
    });
    test("Standard deck matches snapshot", () => {
        const dealer = new CardDealer();

        expect(dealer.deck).toMatchSnapshot();
    });
    test("Accepts a custom deck", () => {
        const dealer = new CardDealer(mockCustomDeck);

        expect(dealer.deck).toEqual(mockCustomDeck);
    });
    test("Deal returns current card index if given no argument", () => {
        const dealer = new CardDealer(mockCustomDeck);

        expect(dealer.deal()).toBe(mockCustomDeck[0]);
    });
    test("Deal returns the amount of card given as an argument", () => {
        const dealer = new CardDealer();

        expect(dealer.deal(10).length).toBe(10);
    });
    test("Deal throws an error if there are not enough cards remaining", () => {
        const dealer = new CardDealer();

        expect(() => {
            dealer.deal(dealer.deck.length + 5);
        }).toThrowError();
    });
    test("Card index increases by amount of cards dealt", () => {
        const dealer = new CardDealer();

        const amount = 5;

        dealer.deal(5);

        expect(dealer._cardIdx).toBe(5);
    });
    test("Shuffle shuffles the deck and returns the class instance", () => {
        const dealer = new CardDealer(mockCustomDeck);

        expect(dealer.shuffle()).toBe(dealer);
        expect(dealer.deck).not.toBe(mockCustomDeck);
    });
    test("Shuffling resets the card index", () => {
        const dealer = new CardDealer();

        dealer.deal(10);

        expect(dealer.shuffle()._cardIdx).toBe(0);
    });
});
