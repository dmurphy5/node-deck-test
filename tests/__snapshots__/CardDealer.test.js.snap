// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDealer Standard deck matches snapshot 1`] = `
Array [
  Object {
    "suit": "Hearts",
    "value": 1,
  },
  Object {
    "suit": "Hearts",
    "value": 2,
  },
  Object {
    "suit": "Hearts",
    "value": 3,
  },
  Object {
    "suit": "Hearts",
    "value": 4,
  },
  Object {
    "suit": "Hearts",
    "value": 5,
  },
  Object {
    "suit": "Hearts",
    "value": 6,
  },
  Object {
    "suit": "Hearts",
    "value": 7,
  },
  Object {
    "suit": "Hearts",
    "value": 8,
  },
  Object {
    "suit": "Hearts",
    "value": 9,
  },
  Object {
    "suit": "Hearts",
    "value": "J",
  },
  Object {
    "suit": "Hearts",
    "value": "Q",
  },
  Object {
    "suit": "Hearts",
    "value": "K",
  },
  Object {
    "suit": "Spades",
    "value": 1,
  },
  Object {
    "suit": "Spades",
    "value": 2,
  },
  Object {
    "suit": "Spades",
    "value": 3,
  },
  Object {
    "suit": "Spades",
    "value": 4,
  },
  Object {
    "suit": "Spades",
    "value": 5,
  },
  Object {
    "suit": "Spades",
    "value": 6,
  },
  Object {
    "suit": "Spades",
    "value": 7,
  },
  Object {
    "suit": "Spades",
    "value": 8,
  },
  Object {
    "suit": "Spades",
    "value": 9,
  },
  Object {
    "suit": "Spades",
    "value": "J",
  },
  Object {
    "suit": "Spades",
    "value": "Q",
  },
  Object {
    "suit": "Spades",
    "value": "K",
  },
  Object {
    "suit": "Clubs",
    "value": 1,
  },
  Object {
    "suit": "Clubs",
    "value": 2,
  },
  Object {
    "suit": "Clubs",
    "value": 3,
  },
  Object {
    "suit": "Clubs",
    "value": 4,
  },
  Object {
    "suit": "Clubs",
    "value": 5,
  },
  Object {
    "suit": "Clubs",
    "value": 6,
  },
  Object {
    "suit": "Clubs",
    "value": 7,
  },
  Object {
    "suit": "Clubs",
    "value": 8,
  },
  Object {
    "suit": "Clubs",
    "value": 9,
  },
  Object {
    "suit": "Clubs",
    "value": "J",
  },
  Object {
    "suit": "Clubs",
    "value": "Q",
  },
  Object {
    "suit": "Clubs",
    "value": "K",
  },
  Object {
    "suit": "Diamonds",
    "value": 1,
  },
  Object {
    "suit": "Diamonds",
    "value": 2,
  },
  Object {
    "suit": "Diamonds",
    "value": 3,
  },
  Object {
    "suit": "Diamonds",
    "value": 4,
  },
  Object {
    "suit": "Diamonds",
    "value": 5,
  },
  Object {
    "suit": "Diamonds",
    "value": 6,
  },
  Object {
    "suit": "Diamonds",
    "value": 7,
  },
  Object {
    "suit": "Diamonds",
    "value": 8,
  },
  Object {
    "suit": "Diamonds",
    "value": 9,
  },
  Object {
    "suit": "Diamonds",
    "value": "J",
  },
  Object {
    "suit": "Diamonds",
    "value": "Q",
  },
  Object {
    "suit": "Diamonds",
    "value": "K",
  },
]
`;
